### ISSUE TITLE
Output latest rclone version with rclone version --check

### ISSUE THREAD

**Issue by jahio** [ncw/rclone#2348]
Currently, rclone reports its existing version when you run `rclone version`. I suggest that if the user specifies any level of verbosity, rclone could output the latest publicly available stable version in that output as well.

First, I'm not suggesting the output of `rclone version` or `rclone --version` change at all - I'm sure somebody's shell scripts somewhere depend on that. Instead, I'm suggesting that if the user passes one or more levels of verbosity, a simple HTTP GET could be performed with something like a 100ms timeout to rclone.org/versions/latest (or somewhere) that just returns a very simple string containing _only_ the version number, nothing else. If you get that response back in < 100ms, display appropriate output.

For example:

```
$ rclone version -v
rclone v1.41
- os/arch: darwin/amd64
- go version: go1.10.1
Latest available via rclone.org: v1.4.9
Update your package manager or visit rclone.org to upgrade to the latest stable release!
```

> I traveled to the future to get the above example, of course, after rclone v1.4.9 was released with this feature. 😃BTW they have hover cars in the future, true story!

Alternatively, if you're on the latest version:

```
$ rclone version -v
rclone v1.41
- os/arch: darwin/amd64
- go version: go1.10.1
Latest available via rclone.org: v1.4.1
Thanks for running the latest stable version of rclone!
```

> This sample from an alternate reality where Hover Boards DO work on water!

And if rclone.org doesn't respond within < 100ms (or whatever's short and reasonable) for some reason? (User offline, internet exploded, clowns ate the server...), just ignore it and do what's already being done right now (omit last two lines).

Optionally this could be put behind a `-vv` flag instead of just one level of verbosity, if that suits others better.

Basically I think it would be nice to have a way to check what's _actually_ running - not what my package manager tells me, which may be wrong or lagging behind - against what's actually released. This could be a very handy feature for such a fast-moving and promising project!

**Comment by jahio** [comment]
Thanks again for your feedback, @ncw. To respond with some thoughts:

> I'm not sure I like the user interface, but I really like the idea behind it.

Thank you! I'm far from insistent on the interface or the output, so feel free to play with the concept all you want. To differentiate, I'm thinking of "interface" as "what I type to get the info", and "output" as "what info gets displayed".

One thing you _could_ do is publish a JSON file with maybe a bit more data in it somewhere (where the aforementioned version file would go, or along-side it). Maybe it looks something like this:

> Admittedly this is just a first crack without much (any) planning so probably not perfect

```json
{
  "versions":
  {
    "latest": 
    {
      "version":"1.4.1",
      "release_date":"2018-MM-DD 13:54:00 UTC",
      "source":"https://github.com/ncw/rclone/releases/tag/v1.41",
      "platforms":
      [
        {
          "macos":
          {
            "files":
            [
              {
                "arch": "x64",
                "download": "https://github.com/ncw/rclone/releases/download/v1.41/rclone-v1.41-osx-amd64.zip",
                "checksums":
                [
                  { "md5":  "..." },
                  { "sha1": "..." }
                ]
              },
              {
                "arch": "i386",
                "download": "https://github.com/ncw/rclone/releases/download/v1.41/rclone-v1.41-osx-386.zip",
                "checksums":
                [
                  { "md5":  "..." },
                  { "sha1": "..." }
                ]    
              }
            ]
          }
        },
        {
          "windows":
          {
            "_comment":"repeat the structure, you get the idea"
          }
        }
      ]
    }
  },
  "previous":
  {
    "_comment":"same as latest, just with appropriate values for that version"
  }
}
```

(Obviously that should be minified without whitespace for production.)

By doing something like this, you've got a canonical, singular source of truth for the _two most recent_ release versions, and by having this, not only can `rclone` fetch it and display whatever the user asks for (notes on that below), **but automation tools can count on this file being available too, so they can script against it and have an always up-to-date singular source of information** so they can automate builds, update installations, etc. however they want.

So now that we have more data in a way that's sort of "universally" useful...

> Examples of my current thinking (unrefined)

The first thing that occurs to me is, perhaps: `rclone version --check [-v|-vv]`

Adding a `--check` flag could be what we need to differentiate it enough so that some one has to _purposefully_ ask for this information before getting it unexpectedly. With `-v`, rclone could output a _little_ more information, or with `-vv` it could give them a _lot_ (as it pertains to the user's platform).

Example concept:

```
$ rclone version --check
   yours:   1.4.1
   latest:  1.4.3
   upgrade:
       https://github.com/ncw/rclone/releases/tag/v1.43

#
# This example is for a MacOS x64 install.
# For other platforms the output would be adjusted accordingly
#

$ rclone version --check -v
   yours:   1.4.1
   latest:  1.4.3 (released 2018-09-22)
   upgrade:
       https://github.com/ncw/rclone/releases/tag/v1.43
   download:
       https://github.com/ncw/rclone/releases/download/v1.43/rclone-v1.43-osx-amd64.zip

#
# An example for a Linux i386 run (for contrast) using -vv
# Here I'm showing just the match of platform in "latest",
# so as not to flood the user with way too much information about all the various
# platform/arch combinations. Note that if the platform you're on is suddenly
# discontinued, this would be a bit of an edge case where we should warn the
# user somehow, maybe with an `exit 99` or something and a big warning in caps
#

$ rclone version --check -vv
   yours:   1.4.1
       platform: linux
       arch: 386
   latest:  1.4.3 (previous: 1.4.2, released 227 days ago)
       platforms: linux 
       archs: [ "i386", "x64" ]
   upgrade:
       https://github.com/ncw/rclone/releases/tag/v1.43
   download:
       https://github.com/ncw/rclone/releases/download/v1.43/rclone-v1.43-linux-386.zip
           checksums:
             md5:  value
             sha1: value
   release metadata (JSON):
       https://rclone.org/versions/latest.json
```

Again, I'm far from insistent on _anything_ here. I'm just throwing what's in my head at you in hopes that it'll spark some creativity in you and/or your team to carry this forward as you see fit. Thanks again for your reply!

**Comment by ncw** [comment]
Thanks for your thoughts.  I like the `--check` parameter to `rclone version` that sounds good!

I can fetch the current version and the current beta release version very easily and I can get the dates of those releases without changing any of the build architecture.

**Comment by ncw** [comment]
I have made a first version of this roughly along the lines of your first suggestion...

https://beta.rclone.org/v1.42-006-gb77a330d-fix-2348-version-check/ (uploaded in 15-30 mins)

What do you think?

Here is the help

### Version

Show the version number, the go version and the architecture.

Eg

    $ rclone version
    rclone v1.41
    - os/arch: linux/amd64
    - go version: go1.10

If you supply the --check flag, then it will do an online check to
compare your version with the latest release and the latest beta.

    $ rclone version --check
    yours:  1.42.0.6
    latest: 1.42          (released 2018-06-16)
    beta:   1.42.0.5      (released 2018-06-17)

Or

    $ rclone version --check
    yours:  1.41
    latest: 1.42          (released 2018-06-16)
      upgrade: https://downloads.rclone.org/v1.42
    beta:   1.42.0.5      (released 2018-06-17)
      upgrade: https://beta.rclone.org/v1.42-005-g56e1e820

**Comment by ncw** [comment]
Oops, try this instead

https://beta.rclone.org/v1.42-006-g5821e7c7-fix-2348-version-check/ (uploaded in 15-30 mins)

**Comment by ncw** [comment]
I've merged this to master now

https://beta.rclone.org/v1.42-132-g4eefd05d/ (uploaded in 15-30 mins)

This will be in [the latest beta](https://beta.rclone.org) from now on and 1.43
